tools/config-gen/params.yaml

```diff
@@ -15,16 +31,308 @@
+  config-path: "metadata-cache.deprecated-stat-cache-capacity"
+  type: "int"
+  usage: >-
+    How many entries can the stat-cache hold (impacts memory consumption). This
+    flag has been deprecated (starting v2.0) and in favor of
+    stat-cache-max-size-mb. For now, the value of stat-cache-capacity will be
+    translated to the next higher corresponding value of stat-cache-max-size-mb
+    (assuming stat-cache entry-size ~= 1640 bytes, including 1400 for positive
+    entry and 240 for corresponding negative entry), if stat-cache-max-size-mb
+    is not set."
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    stat-cache-max-size-mb.
+  default: "20460"
+
+- flag-name: "stat-cache-ttl"
+  config-path: "metadata-cache.deprecated-stat-cache-ttl"
+  type: "duration"
+  usage: >-
+    How long to cache StatObject results and inode attributes. This flag
+    has been deprecated (starting v2.0) in favor of metadata-cache-ttl-secs. For
+    now, the minimum of stat-cache-ttl and type-cache-ttl values, rounded up to
+    the next higher multiple of a second is used as ttl for both stat-cache and
+    type-cache, when metadata-cache-ttl-secs is not set.
+  default: "60s"
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    metadata-cache-ttl-secs.
+
+- flag-name: "type-cache-ttl"
+  config-path: "metadata-cache.deprecated-type-cache-ttl"
+  type: "duration"
+  usage: >-
+    Usage: How long to cache StatObject results and inode attributes. This flag
+    has been deprecated (starting v2.0) in favor of metadata-cache-ttl-secs. For
+    now, the minimum of stat-cache-ttl and type-cache-ttl values, rounded up to
+    the next higher multiple of a second is used as ttl for both stat-cache and
+    type-cache, when metadata-cache-ttl-secs is not set.
+  default: "60s"
+  deprecated: true
+  deprecation-warning: >-
+    This flag has been deprecated (starting v2.0) in favor of
+    metadata-cache-ttl-secs.
+
+- flag-name: "kernel-list-cache-ttl-secs"
+  config-path: "list.kernel-list-cache-ttl-secs"
+  type: "int"
+  usage: >-
+    How long the directory listing (output of ls <dir>) should be cached in the
+    kernel page cache. If a particular directory cache entry is kept by kernel
+    for longer than TTL, then it will be sent for invalidation by gcsfuse on
+    next opendir (comes in the start, as part of next listing) call. 0 means no
+    caching. Use -1 to cache for lifetime (no ttl). Negative value other than -1
+    will throw error.
+
+- flag-name: "http-client-timeout"
+  config-path: "gcs-connection.http-client-timeout"
+  type: "duration"
+  usage: >-
+    The time duration that http client will wait to get response from the
+    server. The default value 0 indicates no timeout.
+
+- flag-name: "max-retry-duration"
+  type: "duration"
+  usage: "This is currently unused."
+  deprecated: true
```

Comment:
@Tulsishah just confirming that we are going to deprecate this flag (in favor of adding max-retry-count flag)?

@kislaykishore we can probably update the warning message accordingly.

Reply:
Yes. We are

Reply:
I'd like to keep the warning as is. We can update the warning once the new flag is added.